bftengine/src/bcstatetransfer/AsyncStateTransferCRE.cpp

```diff
@@ -154,14 +177,31 @@ std::shared_ptr<ClientReconfigurationEngine> CreFactory::create(std::shared_ptr<
   bftClientConf.replicas_master_key_folder_path = std::nullopt;
   std::unique_ptr<ICommunication> comm = std::make_unique<Communication>(msgsCommunicator, msgHandlers);
   bft::client::Client* bftClient = new bft::client::Client(std::move(comm), bftClientConf);
```

Comment:
can we use a smart pointer?

Reply:
I changed this function, but the interfaces should also reflect the ownership model.
Since the interface changes also affect clients, I think that they are out of scope.